src/EnergyPlus/WaterUse.hh

```diff
@@ -271,28 +248,17 @@ namespace WaterUse {
 struct WaterUseData : BaseGlobalStruct
 {
 
-    int numWaterEquipment;
-    int numWaterConnections;
-    bool getWaterUseInputFlag;
-    bool MyEnvrnFlagLocal;
+    int numWaterEquipment = 0;
+    int numWaterConnections = 0;
+    bool getWaterUseInputFlag = true;
+    bool MyEnvrnFlagLocal = true;
     Array1D_bool CheckEquipName;
-    Array1D<WaterUse::WaterEquipmentType> WaterEquipment;
+    EPVector<WaterUse::WaterEquipmentType> WaterEquipment;
     EPVector<WaterUse::WaterConnectionsType> WaterConnections;
 
     void clear_state() override
     {
-        this->numWaterEquipment = 0;
-        this->numWaterConnections = 0;
-        this->getWaterUseInputFlag = true;
-        this->MyEnvrnFlagLocal = true;
-        this->CheckEquipName.deallocate();
-        this->WaterEquipment.deallocate();
-        this->WaterConnections.deallocate();
-    }
-
-    // Default Constructor
-    WaterUseData() : numWaterEquipment(0), numWaterConnections(0), getWaterUseInputFlag(true), MyEnvrnFlagLocal(true)
-    {
+        new (this) WaterUseData();
```

Comment:
Placement new is OK in this member function, but we should be cautious as some folks might try to use it inside constructors to call other constructors, which is frowned upon.